### ISSUE TITLE
pytorch tensor serialization and deserialization

### ISSUE THREAD

**Issue by lzmchina** [bytedeco/javacpp-presets#1240]
hello, I have a problem about Tensor's serialization and deserialization.
I want to transport Tensor in different java application, so how can I serialize Tensor type data  and deserialize it? 
Thanks a lot!

**Comment by HGuillemet** [comment]
The `InputArchive` and `OutputArchive` classes can serialize tensors with methods `write` and `¶ead`.
You can also serialize whole modules with `Module.load` and `Module.save`.
Same for optimizers.

**Comment by lzmchina** [comment]
>  
Are there some example code about InputArchive and OutputArchive?Thanks a lot!

**Comment by lzmchina** [comment]
> > 
> 
> Are there some example code about InputArchive and OutputArchive?Thanks a lot!
for example, I want to serialize a tensor to bytes array and deserialize the bytes array to tensor in another java program

**Comment by HGuillemet** [comment]
Writing and reading files inbetween ?

**Comment by lzmchina** [comment]
> 

not files.in fact, I want to transport Tensor by http, so I should parse Tensor to bytes array and then deserialize it into Tensor

**Comment by lzmchina** [comment]
@HGuillemet How can I realize it?Thanks a lot!

**Comment by HGuillemet** [comment]
We should be able to use `OutputArchive.save_to(WriteFunction)` but it doesn't work. Still under investigation: #1244 .

**Comment by HGuillemet** [comment]
The bug with `OutputArchive.save_to(WriteFunction)` is now fixed in 1.5.8-SNAPSHOT.
You can try to serialize your tensor over http with something like:
```java
  WriteFunction wf = new WriteFunction() {
      @Override
      public long call(Pointer buf, long nbytes) {
        /* send the data chunk in buf over http, encoded somehow */
        return nbytes;
      }
    };

    void serialize(Tensor t) {
          OutputArchive oa = new OutputArchive();
          oa.write("test", t);
          oa.save_to(wf);
     }
```

And the other way around:
```java
  ReadFunction rf = new ReadFunction() {
      @Override
      public long call(long pos, Pointer buf, long nbytes) {
        /* Read nbytes into buf from position pos of the serialized stream */
        return nbytes;
      }
    };
  SizeFunction sf = new SizeFunction() {
      @Override
      public long call() {
        /* Return total length of stream. Returning -1 if unknown seems to work too.*/
      }
    };

 Tensor deserialize() {
    InputArchive ia = new InputArchive();
    ia.load_from(rf, sf);
    Tensor t = new Tensor();
    ia.read("test", t);
    return t;
  }
```

**Comment by lzmchina** [comment]
I get it.Thank u
